### PR TITLE
Use CloseOutputAsync to close client transport WebSocket

### LINKED ISSUE #568: Revisit how we close the webSockets transport on the client
Currently the webSockets transport on the client uses [CloseAsync](https://github.com/aspnet/SignalR/blob/0298868c00c4254ae283cf3669eb7bdb7a341ff4/src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs#L230) to close the webSocket. `CloseAsync` initiates closing the webSocket and waits until the server sends a close frame to the client. `CloseAsync` can block forever if the server never sends the close frame and there is no good way to time out this call (CancellationToken is not respected). A better way of Closing the webSocket is to use `CloseOutputAsync` which initiates closing the websocket but does not wait for the Close frame from the server. We should then try to drain the webSocket by reading until we receive a close frame or a timeout occurs. Only when we finished draining the webSocket we should `Dispose()` the webSocket.

### PR DESCRIPTION
**Pull request by moozzyk**
Fixes #568

### CODE REVIEW

**Comment by davidfowl** [comment]
Whats the deal with the build?

File: src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs [hunk 1]
```diff
@@ -242,7 +242,9 @@ public WebSocketsTransport(ILoggerFactory loggerFactory)
                 if (_webSocket.State != WebSocketState.Closed)
                 {
                     _logger.ClosingWebSocket(_connectionId);
-                    await _webSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, null, CancellationToken.None);
+                    await _webSocket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, null, CancellationToken.None);
+                    // shutdown the transport after a timeout in case the server does not send close frame
+                    _transportCts.CancelAfter(TimeSpan.FromSeconds(5));
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs]: Why not pass this token into CloseOutputAsync?

**Comment by moozzyk** [comment]
IntegrationTesting - all repos where it was used broke due to using an outdated version. AFAICT @JunTaoLuo has [fixed](47d14e5d4545c475f787b8a0efd4b6acd1a13ac0) it.

File: src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs (hunk 1 above)
Comment (moozzyk) [inline src/Microsoft.AspNetCore.Sockets.Client.Http/WebSocketsTransport.cs]: Actually, if the token is cancelled due to an exception in Send we would never try to close the websocket gracefully

**Comment by moozzyk** [comment]
Need to investigate build failures...

**Comment by BrennanConroy** [comment]
Here is a branch I made last month https://github.com/aspnet/SignalR/compare/brecon/websocketsClose
Don't remember why I moved the dispose but perhaps there is a good reason?

**Comment by moozzyk** [comment]
The reason to move Dispose is to prevent from disposing immediately after CloseOutputAsync is closed - this is why CI was not happy. It is a race - it happens to pass locally.

**Comment by moozzyk** [comment]
:up::date:

**Pull request merged**